src/Servers/HttpSys/src/RequestProcessing/Response.cs

```diff
@@ -403,7 +404,12 @@ internal uint ComputeHeaders(long writeCount, bool endOfRequest = false)
 
         // Determine if the connection will be kept alive or closed.
         var keepConnectionAlive = true;
-        if (requestVersion <= Constants.V1_0 // Http.Sys does not support "Keep-Alive: true" or "Connection: Keep-Alive"
+
+        // An HTTP/1.1 server may also establish persistent connections with
+        // HTTP / 1.0 clients upon receipt of a Keep-Alive connection token.
+        // However, a persistent connection with an HTTP/1.0 client cannot make
+        // use of the chunked transfer-coding. From: https://www.rfc-editor.org/rfc/rfc2068#section-19.7.1
+        if ((requestVersion <= Constants.V1_0 && (!requestConnectionKeepAliveSet || responseChunkedSet))
```

Comment:
We should check for Http/0.9 and not set keep-alive in that case too.

Reply:
Also, let's put this addition behind an opt-in feature flag since Http/1.0 + Keep-Alive isn't the most well-defined feature.
See https://github.com/dotnet/aspnetcore/blob/59d4050e7a26326764aa0f008b378f0adbc40efa/src/Caching/StackExchangeRedis/src/RedisCacheOptions.cs#L57 as an example. Probably call it something like "Microsoft.AspNetCore.Server.HttpSys.AllowHttp10KeepAlive"?

Reply:
I don't think this needs to be opt-in since it aligns closer to the current behavior Kestrel and IIS which both already allow HTTP/1.0 keep-alive connections.

Reply:
> We should check for Http/0.9 and not set keep-alive in that case too.

I'm leaning towards not touching Http/0.9 as the change is intended for Http/1.0. Let me know if that makes sense.

> I don't think this needs to be opt-in since it aligns closer to the current behavior Kestrel and IIS which both already allow HTTP/1.0 keep-alive connections.

Agreed. I thought about using feature flag but I didn't go down that path for the same reason.